specs/governance/alligator.md

```diff
@@ -0,0 +1,179 @@
+    - [`afterTokenTransfer`](#aftertokentransfer)
+  - [Getters](#getters)
+    - [`getSubdelegations`](#getsubdelegations)
+    - [`getCheckpoints`](#getcheckpoints)
+    - [`getVotingPower`](#getvotingpower)
+- [Storage](#storage)
+- [Types](#types)
+  - [`SubdelegationRule`](#subdelegationrule)
+  - [`AllowanceType`](#allowancetype)
+  - [`Checkpoint`](#checkpoint)
+- [Backwards Compatibility](#backwards-compatibility)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `Alligator` contract implements subdelegations that can be used by token contracts inheriting from
+[`GovernanceToken`](gov-token.md). Subdelegations enable advanced delegation use cases, such as partial, time-constrained
+& block-based delegations, and relative & fixed allowances.
+
+The `Alligator` contract migrates the delegation state from a token contract to itself through a hook-based approach.
+Specifically, the token contract calls the `Alligator` contract's `afterTokenTransfer` function after a token
+transfer. This enables the `Alligator` contract to consume the hook and update its delegation and checkpoint mappings
+accordingly. If either address involved in the transfer (`_from_` or `_to`) has not been migrated to the `Alligator` contract,
+the contract copies the address' delegation and checkpoint data from the token contract to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `subdelegate`
+
+Allows subdelegation of token voting power to another address with specified subdelegation rules. This function
+is inteded to be called by users.
+
+```solidity
+subdelegate(address _token, address _delegatee, SubdelegationRule _rule)
+```
+
+#### `subdelegate`
+
+Allows subdelegation of token voting power to another address with specified subdelegation rules. This function
+is intended to be called by the token contract.
+
+```solidity
+subdelegate(address _account, address _delegatee, SubdelegationRule _rule)
+```
+
+#### `subdelegateBatched`
+
+Allows batch subdelegation of token voting power to multiple addresses with specified subdelegation rules. This
+function is intended to be called by users.
+
+```solidity
+subdelegateBatched(address _token, address[] _delegatees, SubdelegationRule[] _rules)
+```
+
+Calls the `subdelegate` function for each pair of `_delegatees` address and subdelegation rule.
+
+#### `afterTokenTransfer`
+
+Updates delegation and checkpoint mappings for the address of a token contract after a transfer. This function
+MUST be called by the `_afterTokenTransfer` function in the token contract.
+
+```solidity
+afterTokenTransfer(address _from, address _to, uint256 _amount)
+```
+
+If the `_to` or `_from` addresses have not been migrated, `Alligator` migrates by copying the delegation and checkpoint
```

Comment:
How do you know if the address was migrated?

Reply:
clarified that the contract must check against the `migrated` mapping, and update it after migrating an address.